syntax = "proto3";

import "google/protobuf/empty.proto";

package lesson;

// Lesson message structure
message Lesson {
    int32 id = 1;
    string title = 2;
    int32 course_id = 3;
}

// Requests and responses
message GetLessonRequest {
    int32 id = 1;  // ID of the lesson
}

message GetLessonsByCourseRequest {
    int32 course_id = 1;  // Course ID for fetching lessons
}

message GetAllLessonsResponse {
    repeated Lesson lessons = 1;  // Response with a list of lessons
}

// Common messages (previously from base.proto)
message IdRequest {
    int32 id = 1;
}

message BaseResponse {
    bool success = 1;
    string message = 2;
}

// gRPC service definition
service LessonService {
    rpc GetLessonById(GetLessonRequest) returns (Lesson);
    rpc GetLessonsByCourse(GetLessonsByCourseRequest) returns (GetAllLessonsResponse);
    rpc GetAllLessons(google.protobuf.Empty) returns (GetAllLessonsResponse);
    rpc CreateLesson(Lesson) returns (BaseResponse);
    rpc UpdateLesson(Lesson) returns (BaseResponse);
    rpc DeleteLesson(IdRequest) returns (BaseResponse);
}
